applicationset/controllers/applicationset_controller.go

```diff
@@ -458,13 +463,18 @@ func (r *ApplicationSetReconciler) validateGeneratedApplications(ctx context.Con
 			errorsByIndex[i] = fmt.Errorf("ApplicationSet %s contains applications with duplicate name: %s", applicationSetInfo.Name, app.Name)
 			continue
 		}
-		_, err := r.ArgoAppClientset.ArgoprojV1alpha1().AppProjects(r.ArgoCDNamespace).Get(ctx, app.Spec.GetProject(), metav1.GetOptions{})
-		if err != nil {
-			if apierr.IsNotFound(err) {
-				errorsByIndex[i] = fmt.Errorf("application references project %s which does not exist", app.Spec.Project)
-				continue
+
+		if !argoProjCache.Contains(app.Spec.GetProject()) {
+			_, err := r.ArgoAppClientset.ArgoprojV1alpha1().AppProjects(r.ArgoCDNamespace).Get(ctx, app.Spec.GetProject(), metav1.GetOptions{})
```

Comment:
Is it necessary to implement our own caching layer, or could we just use an informer?

Reply:
If you give me more insights about informer, I can try to use them :+1: 
I started with a cache layer as quick win and because I know which library to use.  But I'm open to use any other solution.

Reply:
Hi, 

I have updated PR to use the local k8s `Client` which seems already managing cache.
Is that you have in mind ? if OK, I will then fix tests.
Thanks

Reply:
Tests are updated.